fpga_interchange/examples/chipdb.cmake

```diff
@@ -0,0 +1,290 @@
+    set(patch_data ${create_patched_device_db_patch_data})
+    set(input_device ${create_patched_device_db_input_device})
+    set(output_target ${create_patched_device_db_output_target})
+
+    get_target_property(input_device_loc ${input_device} LOCATION)
+    set(output_device_file ${CMAKE_CURRENT_BINARY_DIR}/${device}_${patch_name}.device)
+    add_custom_command(
+        OUTPUT ${output_device_file}
+        COMMAND
+            ${PYTHON_EXECUTABLE} -mfpga_interchange.patch
+                --schema_dir ${INTERCHANGE_SCHEMA_PATH}
+                --schema device
+                --patch_path ${patch_path}
+                --patch_format ${patch_format}
+                ${input_device_loc}
+                ${patch_data}
+                ${output_device_file}
+        DEPENDS
+            ${patch_data}
+            ${input_device}
+            ${input_device_loc}
+    )
+
+    add_custom_target(${patch_name}-${device}-device DEPENDS ${output_device_file})
+    set_property(TARGET ${patch_name}-${device}-device PROPERTY LOCATION ${output_device_file})
+
+    if (DEFINED output_target)
+        set(${output_target} ${patch_name}-${device}-device PARENT_SCOPE)
+    endif()
+endfunction()
+
+function(generate_xc7_device_db)
+    # ~~~
+    # generate_xc7_device_db(
+    #    device <common device>
+    #    part <part>
+    # )
+    # ~~~
+    #
+    # Generates a chipdb BBA file, starting from a RapidWright device database which is then patched.
+    # Patches applied:
+    #   - constraints patch
+    #   - luts patch
+    #
+    # The final device target is output in the device_target variable to use in the parent scope
+
+    set(options)
+    set(oneValueArgs device part)
+    set(multiValueArgs)
+
+    cmake_parse_arguments(
+        create_rapidwright_device_db
+        "${options}"
+        "${oneValueArgs}"
+        "${multiValueArgs}"
+        ${ARGN}
+    )
+
+    set(device ${create_rapidwright_device_db_device})
+    set(part ${create_rapidwright_device_db_part})
+
+    create_rapidwright_device_db(
+        device ${device}
+        part ${part}
+        output_target rapidwright_device
+    )
+
+    # Generate constraints patch
+    create_patched_device_db(
+        device ${device}
+        patch_name constraints
+        patch_path constraints
+        patch_format yaml
+        patch_data ${PYTHON_INTERCHANGE_PATH}/test_data/series7_constraints.yaml
+        input_device ${rapidwright_device}
+        output_target constraints_device
+    )
+
+    # Generate lut constraints patch
+    create_patched_device_db(
+        device ${device}
+        patch_name constraints-luts
+        patch_path lutDefinitions
+        patch_format yaml
+        patch_data ${PYTHON_INTERCHANGE_PATH}/test_data/series7_luts.yaml
+        input_device ${constraints_device}
+        output_target constraints_luts_device
+    )
+
+    set(device_target ${constraints_luts_device} PARENT_SCOPE)
```

Comment:
This should be an argument

Reply:
Just like https://github.com/YosysHQ/nextpnr/pull/619/files#diff-96fd1153eb6fb79f848282c1f05b0a3e0f984461c4ec540413bac18f12d00a0fR116-R118

Reply:
Done